File: DoAnChuyenNganh-main/DoAnCoSo/WebDuLichDaLat/Models/Vehicle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebDuLichDaLat.Models
{
    public class Vehicle
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string DriverName { get; set; }

        [StringLength(20)]
        public string LicensePlate { get; set; }

        [Required]
        public double PickupLatitude { get; set; }

        [Required]
        public double PickupLongitude { get; set; }

        [StringLength(200)]
        public string PickupAddress { get; set; }

        [Required]
        public double DropoffLatitude { get; set; }

        [Required]
        public double DropoffLongitude { get; set; }

        [StringLength(200)]
        public string DropoffAddress { get; set; }

        [Required]
        public DateTime DepartureTime { get; set; }

        [Required]
        public int TotalSeats { get; set; }

        [Required]
        public int AvailableSeats { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal CostPerKm { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? FixedPrice { get; set; } // Giá cố định cho chuyến đi (nếu có)

        [StringLength(50)]
        public string VehicleType { get; set; } // "Xe 4 chỗ", "Xe 7 chỗ", "Xe 16 chỗ", etc.

        // Tuyến chính (polyline từ OSRM) - lưu dạng JSON array of [lon, lat]
        [Column(TypeName = "nvarchar(max)")]
        public string? RoutePolyline { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public bool IsActive { get; set; } = true;
    }
}

File: src/main/java/com/example/dalattravel/model/Vehicle.java
package com.example.DaLattravel.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "vehicles")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 100)
    private String driverName;

    @Column(length = 20)
    private String licensePlate;

    private double pickupLatitude;
    private double pickupLongitude;

    @Column(length = 200)
    private String pickupAddress;

    private double dropoffLatitude;
    private double dropoffLongitude;

    @Column(length = 200)
    private String dropoffAddress;

    @Column(nullable = false)
    private LocalDateTime departureTime;

    private int totalSeats;
    private int availableSeats;

    @Column(precision = 18, scale = 2)
    private BigDecimal costPerKm;

    @Column(precision = 18, scale = 2)
    private BigDecimal fixedPrice;

    @Column(length = 50)
    private String vehicleType;

    @Column(columnDefinition = "TEXT")
    private String routePolyline; // Polyline JSON Array of [lon, lat]

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Builder.Default
    private boolean active = true;
}
